### PR TITLE
misc dependency version updates and other minor matters

### PR DESCRIPTION
**Pull request by hydra3333**
just a few suggestions.

Bumped versions of some of the dependencies.
Slightly different handling of .7z files due to different unzipping commandline with package p7zip-full.
Miscellaneous minor debug statements added.
gcc official version 8.1.0

### CODE REVIEW

File: cross_compiler.py [hunk 1]
```diff
@@ -2537,7 +2600,8 @@ def cchdir(self,dir):
 	},
 	'librsvg' : {
 		'repo_type' : 'archive',
-		'url' : 'https://download.gnome.org/sources/librsvg/2.41/librsvg-2.42.3.tar.xz',
+		#'url' : 'https://download.gnome.org/sources/librsvg/2.41/librsvg-2.42.3.tar.xz',
+		'url' : 'https://download.gnome.org/sources/librsvg/2.42/librsvg-2.42.5.tar.xz', # 2018.06.09
```
Comment (DeadSix27) [inline cross_compiler.py:2604]: Please update this to the new format as well:
```
'download_locations' : [ { "url" : "....", "hashes" : [ { "type" : "sha256", "sum" : "*" }, ],
```

File: cross_compiler.py [hunk 2]
```diff
@@ -2025,7 +2056,7 @@ def cchdir(self,dir):
 			'PKGCONFIG' : 'pkg-config',
 		},
 		'depends_on' : [
-			'libffmpeg',
+			'libffmpeg', # 'liblsw', # ... lsw for x264 requires --enable-avresample which is deprecated # ... gave up on 'libgpac'
```
Comment (DeadSix27) [inline cross_compiler.py:2059]: I used to add comments like this as well, but it's kinda ugly, so I stopped, remove those please and its alright. (same goes to the  2018.06.10 *** ones)

File: cross_compiler.py [hunk 3]
```diff
@@ -3626,13 +3683,13 @@ def cchdir(self,dir):
 	'p11-kit' : {
 		'repo_type' : 'archive',
 		'needs_make_install' : False,
-		'url' : 'hhttps://github.com/p11-glue/p11-kit/releases/download/0.23.9/p11-kit-0.23.9.tar.gz',
+		'url' : 'https://github.com/p11-glue/p11-kit/archive/0.23.12.tar.gz', # 2018.06.10 bump version from 0.23.9
```
Comment (DeadSix27) [inline cross_compiler.py:3686]: Also needs new 'download_locations' format.

**Review by DeadSix27 — CHANGES_REQUESTED** [review]
Make sure to update the marked packages to use 'download_locations' and remove all the 2018.*.* comments and its fine

**Comment by hydra3333** [comment]
OK, will look at it tomorrow.  
Thanks for considering the suggestions.

**Comment by DeadSix27** [comment]
Those deps I Just did not actively use so I never bothered to update them, there are probably more deps that need updating though.

**Comment by DeadSix27** [comment]
@hydra3333 no need to close it btw, you can just add a new commit to this branch and update the PR.

**Comment by hydra3333** [comment]
Oh.

**Pull request closed without merging**